<launch>
  <param name="/use_sim_time" value="true" />
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="slam_methods" default="gmapping" doc="slam type [gmapping, cartographer, hector, karto, frontier_exploration]"/>
  <arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>
  <arg name="open_rviz" default="true"/>
<!--   <include file="$(find turtlebot3_gazebo)/launch/empty_world.launch"/> -->

<!--   replace with LOS  -->
<!--   <node pkg="mdis_state_machine" type="connection_check" name="connection_check" args="2"/> -->

  <arg name="ns_0" value="robot_0"/>
  <arg name="ns_1" value="robot_0"/>
  <arg name="ns_2" value="robot_2"/>
  <arg name="ns_3" value="robot_3"/>

<!-- ======================================================================================================================== -->
<!--   <group ns = "$(arg ns_0)"> -->
<!--     <include file="$(find coms)/launch/map_merger.launch"> -->
<!--         <arg name="ns" value="$(arg ns_0)" /> -->
<!--         <arg name="starting_map" value="$(find coms)/tests/test_data/full_map.pgm"/> -->
<!--     </include> -->
<!--   </group> -->
<!--   <node pkg="tf" type="static_transform_publisher" name="world_to_bot_0_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg ns_0)/map 100"/> -->
<!--   <node pkg="tf" type="static_transform_publisher" name="world_to_bot_0_merged_tf_broadcaster"  args="0 0 0 0 0 0 /merged_map /$(arg ns_0)/merged_map 100"/> -->
<!--   <include file="$(find turtlebot3_gazebo)/launch/multi_turtlebot3_single.launch"> -->
<!--     <arg name="ns" value="$(arg ns_0)"/> -->
<!--     <arg name="tb3_x_pos" value="-31"/> -->
<!--     <arg name="tb3_y_pos" value="-8"/> -->
<!--     <arg name="tb3_z_pos" value="0.0"/> -->
<!--     <arg name="tb3_yaw" value="0.0"/> -->
<!--     <arg name="role" value="1"/> -->
<!--     <arg name="parent_name" value="$(arg ns_1)"/> -->
<!--     <arg name="child_name" value=""/> -->
<!--   </include> -->
<!-- ======================================================================================================================== -->
  <group ns = "$(arg ns_1)">
    <include file="$(find coms)/launch/map_merger.launch">
        <arg name="ns" value="$(arg ns_1)" />
<!--         <arg name="starting_map" value="$(find coms)/tests/test_data/full_map.pgm"/> -->
    </include>
  </group>
  <node pkg="tf" type="static_transform_publisher" name="world_to_bot_1_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg ns_1)/map 100"/>
<!--   <node pkg="tf" type="static_transform_publisher" name="world_to_bot_1_merged_tf_broadcaster"  args="0 0 0 0 0 0 /merged_map /$(arg ns_1)/merged_map 100"/> -->
  <include file="$(find turtlebot3_gazebo)/launch/multi_turtlebot3_single.launch">
    <arg name="ns" value="$(arg ns_1)"/>
    <arg name="tb3_x_pos" value="-31"/>
    <arg name="tb3_y_pos" value="-8"/>
    <arg name="tb3_z_pos" value="0.0"/>
    <arg name="tb3_yaw" value="0.0"/>
    <arg name="role" value="2"/>
    <arg name="parent_name" value=""/>
    <arg name="child_name" value="$(arg ns_0)"/>
  </include>
<!-- ======================================================================================================================== -->


  <!-- Start rviz -->
  <group if="$(arg open_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find turtlebot3_gazebo)/rviz/path_following.rviz"/>
  </group>

</launch>